include/crud-posts.php

```diff
@@ -197,10 +197,12 @@ public function set_object_terms( $object_id, $terms, $tt_ids, $taxonomy ) {
 	 */
 	public function wp_insert_post_parent( $post_parent, $post_id ) {
 		$lang = $this->model->post->get_language( $post_id );
+		$parent_post_type = $post_parent > 0 ? get_post_type( $post_parent ) : null;
 		// Dont break the hierarchy in case the post has no language
-		if ( ! empty( $lang ) ) {
+		if ( ! empty( $lang ) && ! empty( $parent_post_type ) && $this->model->is_translated_post_type( $parent_post_type ) ) {
 			$post_parent = $this->model->post->get_translation( $post_parent, $lang );
```

Comment:
`get_translation()` can return `false` if there is no translation:
```suggestion
			$post_parent = (int) $this->model->post->get_translation( $post_parent, $lang );
```

Reply:
🤔 I think we don't care as before because `false` converted to `0` in DB

Reply:
Indeed, it doesn't break anything, but that doesn't mean we have to be lax about types ლ(ಠ益ಠლ